project/FrontEnd/collaborative_science_platform/lib/screens/workspace_page/mobile_workspace_page/widget/mobile_workspace_content.dart

```diff
@@ -0,0 +1,265 @@
+import 'package:collaborative_science_platform/screens/workspace_page/mobile_workspace_page/widget/reference_card.dart';
+import 'package:collaborative_science_platform/screens/workspace_page/mobile_workspace_page/widget/subsection_title.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+
+import '../../../../models/node.dart';
+import '../../../../models/user.dart';
+import '../../../../models/workspaces_page/entry.dart';
+import '../../../../models/workspaces_page/workspace.dart';
+import '../../../../providers/auth.dart';
+import '../../../../utils/lorem_ipsum.dart';
+import '../../../../utils/responsive/responsive.dart';
+import 'contributor_card.dart';
+import 'entry_card.dart';
+
+class MobileWorkspaceContent extends StatefulWidget {
+  final int workspaceId;
+  final bool pending;
+  const MobileWorkspaceContent({
+    super.key,
+    required this.workspaceId,
+    required this.pending,
+  });
+
+  @override
+  State<MobileWorkspaceContent> createState() => _MobileWorkspaceContentState();
+}
+
+class _MobileWorkspaceContentState extends State<MobileWorkspaceContent> {
+  bool isLoading = false;
+  bool error = false;
+  String errorMessage = "";
+
+  Workspace workspaceData = Workspace(
+    workspaceId: 0,
+    workspaceTitle: "workspaceTitle",
+    entries: <Entry>[],
+    status: WorkspaceStatus.workable,
+    numApprovals: 0,
+    contributors: <User>[],
+    pendingContributors: <User>[],
+    references: <Node>[],
+  );
+
+  @override
+  void didChangeDependencies() {
+    getWorkspaceData();
+    super.didChangeDependencies();
+  }
+
+  void getWorkspaceData() {
+    setState(() {
+      isLoading = true;
+    });
+    workspaceData = Workspace(
+      workspaceId: 0,
+      workspaceTitle: "workspaceTitle",
+      entries: <Entry>[
+        Entry(
+          content: getLongLoremIpsum(),
+          entryDate: DateTime.now(),
+          entryId: 1,
+          entryNumber: 1,
+          index: 1,
+          isEditable: false,
+          isFinalEntry: false,
+          isProofEntry: false,
+          isTheoremEntry: true,
+        ),
+        Entry(
+          content: getLongLoremIpsum(2),
+          entryDate: DateTime.now(),
+          entryId: 2,
+          entryNumber: 2,
+          index: 2,
+          isEditable: false,
+          isFinalEntry: false,
+          isProofEntry: true,
+          isTheoremEntry: false,
+        ),
+        Entry(
+          content: getLongLoremIpsum(3),
+          entryDate: DateTime.now(),
+          entryId: 2,
+          entryNumber: 2,
+          index: 2,
+          isEditable: false,
+          isFinalEntry: true,
+          isProofEntry: true,
+          isTheoremEntry: false,
+        ),
+      ],
+      status: WorkspaceStatus.workable,
+      numApprovals: 0,
+      contributors: <User>[
+        // Automatically add the user to the list of contributors
+        if (!widget.pending) Provider.of<Auth>(context).user as User,
```

Comment:
why do we need to add the user to the contributors list? The user should already be included in the contributor list